### DISCUSSION TITLE
Can I use JAX for evaluating hessians of complicated functions?

### DISCUSSION THREAD

**Discussion by infinite-pursuits** [jax-ml/jax#4715] [Q&A]
Hi there,
First time using JAX. Apologies already for the noob questions.
I need to compute hessians for complicated functions. Getting hessians in functional form by hand takes some time. I am thinking if JAX can compute all this for me?
Is there a guarantee that the answer is going to be correct? What is the degree of approximation?

**Comment by georgedahl** [comment]
If you are new to JAX, some of your answers might be in the https://jax.readthedocs.io/en/latest/notebooks/autodiff_cookbook.html

**Reply by mattjj** [reply]
To answer the specific questions:

> Is there a guarantee that the answer is going to be correct?

Yes, insofar as any software can be guaranteed bug-free. JAX differentiation is extensively tested (we have hundreds of thousands of tests). Moreover, you can always check the result yourself by testing against numerical differentiation in random directions.

> What is the degree of approximation?

It's accurate to machine precision.
